content/en/docs/languages/java/configuration.md

```diff
@@ -1,185 +1,297 @@
 ---
 
-The OpenTelemetry SDK provides a working implementation of the API, and can be
-set up and configured in a number of ways. The Java SDK supports most of the
-available [configuration options](/docs/languages/sdk-configuration/). For
-conformance details, see the
-[compliance matrix](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md).
+<!-- markdownlint-disable blanks-around-fences -->
+<?code-excerpt path-base="examples/java/configuration"?>
+
+The [SDK](/docs/languages/java/sdk/) is the built-in reference implementation of
+the [API](/docs/languages/java/instrumentation/), processing and exporting
+telemetry produced by instrumentation API calls. Configuring the SDK to process
+and export appropriately is an essential step to integrating OpenTelemetry into
+an application.
+
+All SDK components have
+[programmatic configuration APIs](#programmatic-configuration). This is the most
+flexible, expressive way to configure the SDK. However, changing configuration
+requires adjusting code and recompiling the application, and there is no
+language interoperability since the API is written in java.
+
+The [zero-code SDK autoconfigure](#zero-code-sdk-autoconfigure) module offers a
+compelling alternative, configuring SDK components based off system properties /
+environment variables, with various extension points for instances where the
+properties are insufficient. **We recommend the zero-code SDK autoconfigure
+module.**
+
+> The [Java agent](/docs/zero-code/java/agent/) automatically configures the SDK
+> using the zero-code SDK autoconfigure module, and uses it in the
+> instrumentation it installs. All autoconfigure content is applicable to Java
+> Agent users.
+
+## Programmatic configuration
+
+The programmatic configuration interface is the set of APIs for constructing
+[SDK](/docs/languages/java/sdk/) components. All SDK components have a
+programmatic configuration API, and all other configuration mechanisms are built
+on top of this API. For example, the
+[autoconfigure environment variable and system property](#environment-variables-and-system-properties)
+configuration interface interprets well-known environment variables and system
+properties into a series of calls to the programmatic configuration API.
+
+While other configuration mechanisms offer more convenience, none offer the
+flexibility of writing code expressing the precise configuration required. When
+a particular capability isn't supported by a higher order configuration
+mechanism, you may have no choice but to use programmatic configuration.
+
+The [SDK components](/docs/languages/java/sdk/#sdk-components) sections
+demonstrate simple programmatic configuration API for key user-facing areas of
+the SDK. For an exhaustive set of the available configuration APIs, consult the
+code.
+
+## Zero-code SDK autoconfigure
```

Comment:
I'm trying to rebrand file configuration as "declarative configuration" here: https://github.com/open-telemetry/opentelemetry-specification/pull/4128

Since SDK autoconfigure combines environment variable / system property configuration AND declarative configuration in one module, calling it "declarative configuration" is probably not a great name.

Maybe just "SDK autoconfigure"? Also open to other ideas..